go/obscuronode/enclave/enclave.go

```diff
@@ -221,9 +226,19 @@ func (e *enclaveImpl) SubmitRollup(rollup common2.ExtRollup) {
 }
 
 func (e *enclaveImpl) SubmitTx(tx common2.EncryptedTx) {
-	t := DecryptTx(tx)
-	e.db.StoreTx(t)
-	e.txCh <- t
+	decryptedTx := DecryptTx(tx)
+	verifySignature(decryptedTx)
+	e.db.StoreTx(decryptedTx)
+	e.txCh <- decryptedTx
+}
+
+// Checks that the L2Tx has a valid signature.
+func verifySignature(decryptedTx L2Tx) {
+	signer := types.NewLondonSigner(big.NewInt(ChainID))
+	_, err := types.Sender(signer, &decryptedTx)
```

Comment:
are you sure this is the api to verify a signature.

I'd suggest creating a small test file where we play with signatres, just to make sure it works.
This feels like quite an important thing to make sure the negative fails.

Also, maybe inject some wrongly signed transactions into the simulation, and add a check to make sure they weren't included in the blockchain.
(I'd suggest to do this test after merging @otherview PRs, as the apis have changed a bit on that)

Reply:
Done. Will add simulation checks once @otherview 's work is in.